File: Year-2023/LeetCode-January-Challenge/BinaryTreePreorderTraversal.java
import java.util.*;
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
class Solution {
    private List<Integer> res = new ArrayList<>();
    private void preorder(TreeNode root){
        if(root != null){
            res.add(root.val);
            preorder(root.left);
            preorder(root.right);
        }
    }
    public List<Integer> preorderTraversal(TreeNode root) {
        preorder(root);
        return res;
    }
}
public class BinaryTreePreorderTraversal {
    public static void main(String[] args) {
        
    }
}

File: Year-2023/LeetCode-January-Challenge/BinaryTreePreorderTraversal.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}
var preorderTraversal = function(root) {
    let res = [];
    const preorder = (root) => {
        if(root){
            res.push(root.val);
            preorder(root.left);
            preorder(root.right);
        }
    }
    preorder(root);
    return res;
};